File: crypto/src/main/java/io/oversec/one/crypto/encoding/pad/AbstractRangePadder.java
package io.oversec.one.crypto.encoding.pad;

import android.content.Context;

public abstract class AbstractRangePadder extends AbstractPadder {
    static final int MAX_TAIL_CHARS = 8;


    AbstractRangePadder(Context ctx) {
        super(ctx);
    }


    @Override
    public void reset() {

    }

    @Override
    public char getNextPaddingChar() {

        return (char) (getFirstChar() + Math.random() * (getLastChar() - getFirstChar()));
    }

    abstract char getFirstChar();

    abstract char getLastChar();

    @Override
    String tail() {
        int t = (int) (MAX_TAIL_CHARS * Math.random());
        StringBuilder sb = new StringBuilder(t);
        for (int i = 0; i < t; i++) {
            sb.append(getNextPaddingChar());
        }
        return sb.toString();
    }
}

File: crypto/src/main/java/io/oversec/one/crypto/encoding/pad/AbstractRangePadder.kt
package io.oversec.one.crypto.encoding.pad

import android.content.Context

abstract class AbstractRangePadder internal constructor(ctx: Context) : AbstractPadder(ctx) {

    protected abstract val firstChar: Char
    protected abstract val lastChar: Char

    override fun reset() {}

    override val nextPaddingChar: Char
        get() = (firstChar.toDouble() + Math.random() * (lastChar - firstChar)).toChar()

    override fun tail(): String {
        val t = (MAX_TAIL_CHARS * Math.random()).toInt()
        val sb = StringBuilder(t)
        for (i in 0 until t) {
            sb.append(nextPaddingChar)
        }
        return sb.toString()
    }

    companion object {
        const val MAX_TAIL_CHARS = 8
    }
}
